nixos/modules/programs/xonsh.nix

```diff
@@ -45,7 +45,23 @@ in
 
   config = mkIf cfg.enable {
 
-    environment.etc.xonshrc.text = cfg.config;
+    environment.etc.xonshrc.text = ''
+      # /etc/xonshrc: DO NOT EDIT -- this file has been generated automatically.
+
+
+      if not ''${...}.get('__NIXOS_SET_ENVIRONMENT_DONE'):
+          # The NixOS environment and thereby also $PATH
+          # haven't been fully set up at this point. But
+          # `source-bash` below requires `bash` to be on $PATH,
+          # so add one entry where it can be found for sure:
+          $PATH.add('/run/current-system/sw/bin')
+
+          # Source the NixOS environment config.
```

Comment:
I would have wanted to put a colon here instead of a period, but can't due to xonsh/xonsh#3502.

Reply:
This is a very strange bug.

Reply:
Indeed :stuck_out_tongue: